raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1751,6 +1768,7 @@ private FetchRequestData buildFetchRequest() {
         });
         return request
             .setMaxWaitMs(fetchMaxWaitMs)
+            .setClusterId(clusterId.toString())
```

Comment:
One small detail which is probably ok. The clusterId field in the fetch schema is not currently marked as ignorable. That should be ok since it is only used in the raft implementation which can guarantee that we will have version 12 and above. On the other hand, I don't see any harm making the field ignorable since we are accepting a null value anyway. Is it worth changing that?